go/vt/vttablet/grpcqueryservice/server.go

```diff
@@ -422,8 +422,8 @@ func (q *query) ReserveBeginExecute(ctx context.Context, request *querypb.Reserv
 	)
 	state, result, err := q.server.ReserveBeginExecute(ctx, request.Target, request.PreQueries, request.PostBeginQueries, request.Query.Sql, request.Query.BindVariables, request.Options)
 	if err != nil {
-		// if we have a valid reservedID, return the error in-band
-		if state.ReservedID != 0 {
+		// if we have a valid reservedID or transactionID, return the error in-band
+		if state.TransactionID != 0 || state.ReservedID != 0 {
```

Comment:
Do we have a test for this change?

Reply:
This can be tested through e2e only and we had 6 failing test which are now passing after the change.